sklearn/metrics/classification.py

```diff
@@ -112,6 +112,20 @@ def _weighted_sum(sample_score, sample_weight, normalize=False):
         return sample_score.sum()
 
 
+def _check_labels_subset(present_labels, labels, allow_label_subset):
+    present_labels_ = set(present_labels)
+    labels_ = set(labels)
+
+    if present_labels_ - labels_:
+        if not allow_label_subset:
+            warnings.warn("The labels argument %s was a subset of the present "
```

Comment:
I still don't get why we really want this. In what case does raising an error here help the user? To tell them that they might have done something wrong when they manually entered a set of labels? I think that's a case where making a mistake should return the corresponding bad results. When a user explicitly sets something, they are responsible for it. When they do not, we are responsible for making decent default behaviour; which is what this PR is all about.

For P/R/F and cohen's kappa, we explicitly suggest using `labels` to limit the set of labels. In such a case why should cross validation reach the scoring phase and then blow up because the user failed to set some verbose "I followed the instructions correctly" flag?